monkey/monkey_island/cc/services/attack/attack_config.py

```diff
@@ -0,0 +1,168 @@
+        :param attack_techniques: ATT&CK techniques dict. Format: {'T1110': True, ...}
+        :param monkey_config: Monkey island's configuration
+        :param monkey_schema: Monkey configuration schema
+        """
+        for key, definition in monkey_schema['definitions'].items():
+            for array_field in definition['anyOf']:
+                # Check if current array field has attack_techniques assigned to it
+                if 'attack_techniques' not in array_field:
+                    continue
+                try:
+                    should_remove = not AttackConfig.should_enable_field(array_field['attack_techniques'], attack_techniques)
+                except KeyError:
+                    # Monkey schema field contains not yet implemented technique
+                    continue
+                # If exploiter's attack technique is disabled, disable the exploiter/scanner/PBA
+                AttackConfig.r_alter_array(monkey_config, key, array_field['enum'][0], remove=should_remove)
+
+    @staticmethod
+    def set_booleans(attack_techniques, monkey_config, monkey_schema):
+        """
+        Sets boolean type fields, like "should use mimikatz?" in monkey's config according to ATT&CK matrix
+        :param attack_techniques: ATT&CK techniques dict. Format: {'T1110': True, ...}
+        :param monkey_config: Monkey island's configuration
+        :param monkey_schema: Monkey configuration schema
+        """
+        for key, value in monkey_schema['properties'].items():
+            AttackConfig.r_set_booleans([key], value, attack_techniques, monkey_config)
+
+    @staticmethod
+    def r_set_booleans(path, value, attack_techniques, monkey_config):
+        """
+        Recursively walks trough monkey configuration (DFS) to find which boolean fields needs to be set and sets them
+        according to ATT&CK matrix.
+        :param path: Property names that leads to current value. E.g. ['monkey', 'system_info', 'should_use_mimikatz']
+        :param value: Value of config property
+        :param attack_techniques: ATT&CK techniques dict. Format: {'T1110': True, ...}
+        :param monkey_config: Monkey island's configuration
+        """
+        if isinstance(value, dict):
+            dictionary = {}
+            # If 'value' is a boolean value that should be set:
+            if 'type' in value and value['type'] == 'boolean' and 'attack_techniques' in value:
+                try:
+                    AttackConfig.set_bool_conf_val(path,
+                                                   AttackConfig.should_enable_field(value['attack_techniques'],
+                                                                                    attack_techniques),
+                                                   monkey_config)
+                except KeyError:
+                    # Monkey schema has a technique that is not yet implemented
```

Comment:
I think that should_enable_field raising KeyError is unintuitive, and the way that the functions 'should_enable_field', 'set_arrays' and 'r_set_booleans' work around this don't justify the current way it is.
My suggestion is as follows:
1. should_enable_field assumes 'field_techniques' is valid and not None. The if-statement resulting in raising KeyError should be removed
2. On set_arrays check if `'attack_techniques' in array_field`, only if it is, proceed with should_remove and r_alter_array
3. On r_set_booleans take a similar approach. Check if `'attack_techniques' in value`, only if it is, call should_enable_field and set_bool_conf_val

These changes make the code more readable, and replace an expected-exception design with an if-statement design, which is generally more recommended